misc/build-web-index

```diff
@@ -17,33 +17,45 @@
 # detect when a file lacks such a title and fail loudly, instead of just
 # omitting that file from the index.
 
-set -e
+import re
+import sys
 
-# The non-quotation is intentional: I am trying to get bash to split on words so
-# that `grep` takes the arguments as separate even though they were passed as
-# one thing to this script.
-docs=$1
-examples=$2
-libraries=$3
+def file_block(files):
+  for fname in files:
+    if fname.startswith("doc/"):
+      link_name = fname[len("doc/"):]
+    else:
+      link_name = fname
+    with open(fname, 'r') as f:
+      found_title = False
+      for line in f.readlines():
```

Comment:
Is there any benefit/need to search through the whole file? I would expect that the title should be the first thing that appears in it. Otherwise, we might treat a heading placed halfway as a title, but that seems wrong.

Reply:
Changed to just check the first line.